sdk/cosmos/azure-cosmos/samples/diagnostics_handler_sample.py

```diff
@@ -109,3 +110,74 @@ def should_log(self, **kwargs):
 for item in items:
     container.delete_item(item=str(item['id']), partition_key=[str(item['State']), str(item['City'])])
 
+# Sample usage of using logging filters for diagnostics filtering
+# The same filter parameters that are found in the cosmos_diagnostics_handler can be used in the logging filter
+class CosmosStatusCodeFilter(logging.Filter):
+    def filter(self, record):
+        return hasattr(record, 'status_code') and record.status_code >= 400
```

Comment:
This code seems strange - what is the type of record here and why would it have a status_code attribute?

Reply:
It is a logger Record, which is where we add diagnostics related information to the logs so that they can be filtered. In this update to diagnostics we are passing in extra attributes to the log record each time we log something. But those attributes don't exist by default. Technically you could also just make a function and pass it as a log filter but I inherited from logging.Filter to make this filter as a sample on how to use it with the knowledge that the log record will contain "status_code", "duration", etc.